src/components/side-panel/layer-panel/layer-configurator.js

```diff
@@ -101,9 +104,20 @@ export default class LayerConfigurator extends Component {
           collapsible
         >
           {layer.config.colorField ? (
-            <ColorRangeConfig {...visConfiguratorProps} />
+            <ColorRangeConfig
+              {...visConfiguratorProps}
+              customPalette={this.props.customPalette}
```

Comment:
These 4 props have been repeatedly added way too many times. Can you find a way to only add them once?

Reply:
refactored